src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,295 @@
+     */
+    public static Map<String, String> excludeRealtime(Map<String, String> params) {
+        Map<String, String> result = new HashMap<>(params);
+        result.put(IGNORE_REALTIME_UPDATES_PARAM, "true");
+        return result;
+    }
+
+    /**
+     * Checks that, for each query provided, an itinerary exists.
+     * @return An object with a map of results and summary of itinerary existence.
+     */
+    public static ItineraryExistence checkItineraryExistence(MonitoredTrip trip, boolean checkAllDays, boolean sortDates) throws URISyntaxException {
+        // Get queries to execute by date.
+        Map<ZonedDateTime, String> queriesByDate = ItineraryUtils.getItineraryExistenceQueries(trip, checkAllDays);
+        // TODO: Consider multi-threading?
+        Map<ZonedDateTime, Itinerary> datesWithMatchingItineraries = new HashMap<>();
+
+        Collection<Map.Entry<ZonedDateTime, String>> entriesToIterate = sortDates
+            ? queriesByDate.entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toList())
+            : queriesByDate.entrySet();
+
+        for (Map.Entry<ZonedDateTime, String> entry : entriesToIterate) {
+            ZonedDateTime date = entry.getKey();
+            String params = entry.getValue();
+
+            OtpDispatcherResponse response = OtpDispatcher.sendOtpPlanRequest(params);
+            TripPlan plan = response.getResponse().plan;
+            if (plan != null && plan.itineraries != null) {
+                for (Itinerary itinerary: plan.itineraries) {
+                    // If a matching itinerary is found, save the date with the matching itinerary.
+                    // The matching itinerary will replace the original trip.itinerary.
+                    if (itinerariesMatch(trip.itinerary, itinerary)) {
+                        datesWithMatchingItineraries.put(date, itinerary);
+                    }
+                }
+            }
+        }
+        return new ItineraryExistence(queriesByDate.keySet(), datesWithMatchingItineraries);
+    }
+
+    /**
+     * Overload used in normal conditions.
+     */
+    public static ItineraryExistence checkItineraryExistence(MonitoredTrip trip, boolean checkAllDays) throws URISyntaxException {
+        return checkItineraryExistence(trip, checkAllDays,false);
+    }
+
+    /**
+     * TODO: fully implement
+     */
+    public static boolean itineraryIsSavable(Itinerary itinerary) {
+        return true;
+    }
+
+    /**
+     * Returns true if the itineraries match for the purposes of trip monitoring.
+     *
+     * @param previousItinerary The original itinerary that others are compared against.
+     * @param newItinerary A new itinerary that might match the previous itinerary.
+     */
+    public static boolean itinerariesMatch(Itinerary previousItinerary, Itinerary newItinerary) {
+        // make sure either itinerary is savable before continuing
```

Comment:
I think the true intent is for the comment (and `itineraryIsSavable`) is whether an itinerary can be monitored.

Reply:
Changed comment in 52d438aad1badb194c212a736a16d72aff3da526